server/app/utils.py

```diff
@@ -405,3 +412,47 @@ def check_user(user_key):
         user_key = user_key.id()
 
     deferred.defer(deferred_check_user, user_key)
+
+######################
+# Autograder actions #
+######################
+
+def add_taskqueue(submission):
+    q = taskqueue.Queue("pull-queue")
+    backup = submission.backup.get()
+    submission_contents = backup.get_messages().get("file_contents")
+    sub_id = str(submission.key.id())
+
+    tasks = []
+    tasks.append(taskqueue.Task( payload = str(submission_contents), method = "PULL", tag = sub_id))
+    q.add(tasks)
+
+def add_all_taskqueue(course, assign_key):
+    q = taskqueue.Queue("pull-queue")
+    students = course.students()  
+    final_submissions = [student.get().get_final_submission(assign_key) for student in students]
+    submissions = [final.submission.get() for final in final_submissions]
+    backups = [(sub.backup.get(), str(sub.key.id())) for sub in submissions]
+
+    tasks = []
+    for backup, sub_id in backups:
+        submission_contents = backup.get_messages().get("file_contents")
+        tasks.append(taskqueue.Task( payload = submission_contents, method = "PULL", tag = sub_id))
+    q.add(tasks)
+
+def lease_tasks(): 
```

Comment:
Can someone comment about when/why this gets called and what role it has in the whole architecture?